pkg/apic/provisioning/credentials.go

```diff
@@ -0,0 +1,98 @@
+package provisioning
+
+import "fmt"
+
+const credentialTypeSetError = "can not set credential as %s as its already set as another %s"
+
+// Credential - holds the details about the credential to send to encrypt and send to platform
+type Credential struct {
```

Comment:
This should probably be an interface and an internal struct should hold the values that gets populated by the builder and builder returns to the provisioning interface so that the callback can return it back to agent SDK

Reply:
Made this private and created a Credential interface above it